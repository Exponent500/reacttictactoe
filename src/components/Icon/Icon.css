.svg-container.white {
    color: white;
}