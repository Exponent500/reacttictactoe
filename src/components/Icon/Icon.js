import React from 'react';

const Icon = ({ type }) => {
    return (
        <div>
            <svg width='1em' height='1em'>
                <use xlinkHref={ `#${ type }` }></use>
            </svg>
        </div>
    )
}

export default Icon;