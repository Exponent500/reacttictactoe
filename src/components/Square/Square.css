.square {
    height: 100px;
    width: 100px;
    line-height: 100px;
    cursor: pointer;
    font-size: 75px;
}

.marked {
    cursor: not-allowed;
}