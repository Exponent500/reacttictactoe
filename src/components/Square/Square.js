import React from 'react';
import classNames from 'classnames';

import './Square.css';
import Icon from '../Icon/Icon';

const Square = ({ value, onClick }) => {
        const squareClass = classNames({
            square: true,
            'marked': value
        });

        return (
            <div className={squareClass} onClick={onClick}>
                <div className="square-value">
                    <Icon type={value}></Icon>
                </div>
            </div>
        );
}

export default Square;