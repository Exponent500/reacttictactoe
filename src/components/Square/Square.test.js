import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Square from './Square';
import Icon from '../Icon/Icon';

let wrapped;

beforeEach( () => {
    wrapped = shallow(
        <Square value='mockValue'></Square>
    ); 
});

describe('Square', () => {
    it('should render correctly', () => {
        expect(shallowToJson(wrapped)).toMatchSnapshot();
    });
    it('should include an Icon component', () => {
        expect(wrapped.find(Icon).length).toEqual(1);
    })
});
