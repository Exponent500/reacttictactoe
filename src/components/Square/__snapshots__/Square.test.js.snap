// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square should render correctly 1`] = `
<div
  className="square marked"
>
  <div
    className="square-value"
  >
    <Icon
      type="mockValue"
    />
  </div>
</div>
`;
